src/titan_db_test.cc

```diff
@@ -606,6 +607,40 @@ TEST_F(TitanDBTest, BlobFileCorruptionErrorHandling) {
 }
 #endif  // !NDEBUG
 
+TEST_F(TitanDBTest, BlobRunModeBasic) {
+  Open();
+  const uint64_t kNumEntries = 100;
+  std::map<std::string, std::string> data;
+  for (uint64_t i = 1; i <= kNumEntries; i++) {
+    Put(i, &data);
+  }
+  ASSERT_EQ(kNumEntries, data.size());
+  VerifyDB(data);
+  Flush();
+  VerifyDB(data);
+
+  std::unordered_map<std::string, std::string> opts;
+  opts["blob-run-mode"] = "read-only";
+  db_->SetOptions(opts);
+  for (uint64_t i = kNumEntries + 1; i <= kNumEntries * 2; i++) {
+    Put(i, &data);
+  }
+  ASSERT_EQ(kNumEntries * 2, data.size());
+  VerifyDB(data);
+  Flush();
+  VerifyDB(data);
```

Comment:
You should verify:
1. The number of blob files is not growing
2. value type for key 100-199 are kTypeValue (you can use this helper function: https://github.com/facebook/rocksdb/blob/master/include/rocksdb/utilities/debug.h#L39)